drivers/syslog/Kconfig

```diff
@@ -375,4 +377,10 @@ config SYSLOG_IOCTL
 		commands are SYSLOGIOC_SETFILTER/SYSLOGIOC_GETCHANNELS, which can be
 		used to set the enable status of different channels
 
+config SYSLOG_REGISTER
+	bool "Register syslog channel support"
```

Comment:
should we remove prompt string and change `depend on` to `select`

Reply:
Done